backend/default.nix

```diff
@@ -0,0 +1,44 @@
+{ haskell-nix, runCommand, lib, hlint }:
+let
+  ignoreFilter = path: type:
+    let
+      inherit (lib) removePrefix hasPrefix hasSuffix;
+      relPath = removePrefix (toString ./. + "/") (toString path);
+      baseName = baseNameOf relPath;
+    in
+      !(
+        baseName == ".gitignore" ||
+        hasPrefix "resources" relPath ||
+        hasSuffix ".md" baseName ||
+        hasSuffix ".nix" baseName
+      );
+
+  # Set a constant name for the src closure
+  source = let
```

Comment:
AFAIU this `source` thing and the above `ignoreFilter` are only used to run `hlint`, I see it used only in `hlint-html` (and btw, does `${source}` there refer to this `source` or something else?). Is it indeed needed only for `hlint` or is it implicitly used somewhere else? I just Ctrl-F'ed `source`.

I have an alternative solution: just call `hlint` from `pipeline.yaml` using `nix run`. Pros:
1. No need to have all this extra code in this file (`source`, `ignoreFilter`). Though maybe I am wrong and it's needed for something else as well.
2. Less steps involved, easier to understand what's going on, easier to find what arguments are passed and where to change them.

AFAIU there is one problem: we don't know how to make `nix run` work with flakes. If that's the case and we don't figure it out quickly, I am okay with changing it later.